ROADMAP.md

```diff
@@ -0,0 +1,33 @@
+
+### How to add an item to the roadmap?
+One of the most important aspects in any open source community is the concept of proposals. Large changes to the codebase and / or new features should be preceded by a [proposal](https://github.com/vmware-tanzu/velero/blob/master/GOVERNANCE.md#proposal-process) in our repo. 
+For smaller enhancements, you can open an issue to track that initiative or feature request.
+We work with and rely on community feedback to focus our efforts to improve Velero and maintain a healthy roadmap.
+
+### Current Roadmap
+The following table includes the current roadmap for Velero. If you have any questions or would like to contribute to Velero, please attend a [community meeting](https://velero.io/community/) to discuss with our team. If you don't know where to start, we are always looking for contributors that will help us reduce technical, automation, and documentation debt.
+Please take the timelines & dates as proposals and goals. Priorities and requirements change based on community feedback, roadblocks encountered, community contributions, etc. If you depend on a specific item, we encourage you to attend community meetings to get updated status information, or help us deliver that feature by contributing to Velero.
+
+`Last Updated: May 2020`
+
+|Theme|Description|Timeline|
+|--|--|--|
+|Restic Improvements|Introduce improvements in annotating resources for Restic backup|August 2020|
+|Extensibility|Add restore hooks for enhanced recovery scenarios|August 2020|
+|CSI|Continue improving the CSI snapshot capabilities and participate in the upstream K8s CSI community|Long running (dependent on CSI working group)|
+|Backup/Restore|Improvements to long-running operations, including retry logic|August 2020|
+|Restic Improvements|Improve the use of Restic in Velero and offer stable support|Dec 2020|
+|Self-Service & Multitenancy|Reduce friction by enabling developers to backup their namespaces via self-service. Introduce a Velero multi-tenancy model, enabling owners of namespaces to backup and restore within their access scope|Dec 2020|
+|Backup/Restore|Better backup and restore semantics for certain Kubernetes resources like stateful sets, operators|Dec 2020|
+|Perf & Scale|Introduce a scalable model by using a worker pod for each backup/restore operation and improve operations|Mar 2021|
+|Application Consistency|Offer blueprints for backing up and restoring popular applications|Mar 2021|
+|Backup/Restore|Data only backup and restore|Mar 2021|
+|Backup/Restore|Introduce the ability to overwrite existing objects during a restore|Mar 2021|
+|Backup/Restore|What-if dry run for backup and restore|Mar 2021|
```

Comment:
Should we add support for per BSL credentials or having a mapping of BSL to credentials?
Combination of #1065 and #1655